Compile problem on Mac/brew for ESP-IDF components

### The problem

I have esphome on my development machine (a MBP with M1 CPU). This used to work well, the last time I set up a sensor node using 2025.7.x it went smoothly. When I tried it with the 2025.9.x version, it did fail on me - and it still won't work with the current 2025.10.2 version. It always complains about

`fatal: not a git repository (or any of the parent directories): .git`

and basically complains, that it can't find the IDF component manager

```
CMake Error at /Users/dominik/.platformio/packages/framework-espidf/tools/cmake/build.cmake:620 (message):
  /Users/dominik/.platformio/penv/.espidf-5.4.2/bin/python: Error while
  finding module specification for 'idf_component_manager.prepare_components'
  (ModuleNotFoundError: No module named 'idf_component_manager')

Call Stack (most recent call first):
  /Users/dominik/.platformio/packages/framework-espidf/tools/cmake/project.cmake:718 (idf_build_process)
  CMakeLists.txt:3 (project)
```

when it sets up the IDF toolchain and especially the ESP-IDF's Python dependencies. I also got this error on my home assistant machine briefly (that's running in an ESXI VM) - but the latest update to 2025.10.2 did fix that (as well as the possibility to compile there and download the results without getting a 500 error).

I'm not sure if this is a Mac/Homebrew problem or an esphome problem, maybe you can help me to investigate that further. I tried to delete the .platformio folder to force esphome to download current versions, but that did not change anything.

### Which version of ESPHome has the issue?

2025.10.2

### What type of installation are you using?

pip

### What platform are you using?

ESP32

### Component causing the issue

ESP-IDF's Python dependencies

### YAML Config

```yaml
esphome:
  project:
    name: xx.sensor_platform
    version: 0.1-beta
  name: testsensor

esp32:
  board: esp32-poe
  framework:
    type: esp-idf

wifi:
  networks:
    - ssid: xx_iot
      password: "..."

text_sensor:
  - platform: wifi_info
    ip_address:
      name: IP-Addresse
      icon: mdi:ip-network
      entity_category: diagnostic
      update_interval: 1s
    ssid:
      name: WLAN SSID
      icon: mdi:access-point
      entity_category: diagnostic
      update_interval: 1s

sensor:
  - platform: wifi_signal
    id: testsensor_wifi_rssi
    name: WLAN RSSI
    icon: mdi:wifi
    update_interval: 60s
    entity_category: diagnostic
    device_class: signal_strength
    force_update: false
    unit_of_measurement: dBm
    accuracy_decimals: 0
    state_class: measurement

  - platform: bme280_i2c
    address: 0x76
    temperature:
      id: testsensor_bme280_temperature
      name: Temperatur BME280
      icon: mdi:thermometer
      unit_of_measurement: °C
      device_class: temperature
      state_class: measurement
      accuracy_decimals: 1
      disabled_by_default: false
      force_update: false
      oversampling: 16X
    humidity:
      id: testsensor_bme280_humidity
      name: Luftfeuchtigkeit BME280
      icon: mdi:water-percent
      unit_of_measurement: '%'
      device_class: humidity
      state_class: measurement
      accuracy_decimals: 0
      disabled_by_default: false
      force_update: false
      oversampling: 16X
    pressure:
      id: testsensor_bme280_pressure
      name: Luftdruck BME280
      icon: mdi:gauge
      unit_of_measurement: hPa
      device_class: atmospheric_pressure
      state_class: measurement
      accuracy_decimals: 0
      disabled_by_default: false
      force_update: false
      oversampling: 16X
    iir_filter: 'OFF'
    update_interval: 10s

i2c:
  - id: i2c_uext
    sda: 13
    scl: 16
    scan: true

logger:

api:
  encryption:
    key: ...
ota:
  - platform: esphome
    password: ...
```

### Anything in the logs that might be useful for us?

```txt
❯ rm -rf .platformio
❯ esphome compile /Users/dominik/Documents/_dev/SmartHome/HomeAssistant/_repository/esphome/sensortest.yaml
INFO ESPHome 2025.10.2
INFO Reading configuration /Users/dominik/Documents/_dev/SmartHome/HomeAssistant/_repository/esphome/sensortest.yaml...
INFO Generating C++ source...
INFO Compiling app...
************************************************************************************************************************
If you like PlatformIO, please:
- star it on GitHub > https://github.com/platformio/platformio-core
- follow us on LinkedIn to stay up-to-date on the latest project news > https://www.linkedin.com/company/platformio/
- try PlatformIO IDE for embedded development > https://platformio.org/platformio-ide
************************************************************************************************************************

Processing sensortest (board: esp32-poe; framework: espidf; platform: https://github.com/pioarduino/platform-espressif32/releases/download/54.03.21-2/platform-espressif32.zip)
------------------------------------------------------------------------------------------------------------------------
Platform Manager: Installing https://github.com/pioarduino/platform-espressif32/releases/download/54.03.21-2/platform-espressif32.zip
INFO Installing https://github.com/pioarduino/platform-espressif32/releases/download/54.03.21-2/platform-espressif32.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Platform Manager: espressif32@54.3.21 has been installed!
INFO espressif32@54.3.21 has been installed!
INFO tool-esp_install not installed, installing version https://github.com/pioarduino/esp_install/releases/download/v5.3.1/esp_install-v5.3.1.zip
INFO Installing tool-esp_install version https://github.com/pioarduino/esp_install/releases/download/v5.3.1/esp_install-v5.3.1.zip
Tool Manager: Installing https://github.com/pioarduino/esp_install/releases/download/v5.3.1/esp_install-v5.3.1.zip
INFO Installing https://github.com/pioarduino/esp_install/releases/download/v5.3.1/esp_install-v5.3.1.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: tool-esp_install@5.3.1 has been installed!
INFO tool-esp_install@5.3.1 has been installed!
INFO tool-esp_install successfully installed and verified
INFO Package configuration completed successfully
Tool Manager: Installing https://github.com/pioarduino/esp-idf/releases/download/v5.4.2/esp-idf-v5.4.2.zip
INFO Installing https://github.com/pioarduino/esp-idf/releases/download/v5.4.2/esp-idf-v5.4.2.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: framework-espidf@3.50402.0 has been installed!
INFO framework-espidf@3.50402.0 has been installed!
Tool Manager: Installing https://github.com/pioarduino/registry/releases/download/0.0.1/xtensa-esp-elf-14.2.0_20241119.zip
INFO Installing https://github.com/pioarduino/registry/releases/download/0.0.1/xtensa-esp-elf-14.2.0_20241119.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: toolchain-xtensa-esp-elf@14.2.0+20241119 has been installed!
INFO toolchain-xtensa-esp-elf@14.2.0+20241119 has been installed!
Tool Manager: Installing https://github.com/pioarduino/registry/releases/download/0.0.1/esptoolpy-v5.0.2.zip
INFO Installing https://github.com/pioarduino/registry/releases/download/0.0.1/esptoolpy-v5.0.2.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: tool-esptoolpy@5.0.2 has been installed!
INFO tool-esptoolpy@5.0.2 has been installed!
Tool Manager: Installing https://github.com/pioarduino/registry/releases/download/0.0.1/mklittlefs-3.2.0.zip
INFO Installing https://github.com/pioarduino/registry/releases/download/0.0.1/mklittlefs-3.2.0.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: tool-mklittlefs@3.2.0 has been installed!
INFO tool-mklittlefs@3.2.0 has been installed!
Tool Manager: Installing https://github.com/pioarduino/registry/releases/download/0.0.1/cmake-3.30.2.zip
INFO Installing https://github.com/pioarduino/registry/releases/download/0.0.1/cmake-3.30.2.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: tool-cmake@3.30.2 has been installed!
INFO tool-cmake@3.30.2 has been installed!
Tool Manager: Installing https://github.com/pioarduino/registry/releases/download/0.0.1/esp-rom-elfs-20241011.zip
INFO Installing https://github.com/pioarduino/registry/releases/download/0.0.1/esp-rom-elfs-20241011.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: tool-esp-rom-elfs@2024.10.11 has been installed!
INFO tool-esp-rom-elfs@2024.10.11 has been installed!
Tool Manager: Installing https://github.com/pioarduino/registry/releases/download/0.0.1/ninja-1.13.1.zip
INFO Installing https://github.com/pioarduino/registry/releases/download/0.0.1/ninja-1.13.1.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: tool-ninja@1.13.1 has been installed!
INFO tool-ninja@1.13.1 has been installed!
Tool Manager: Installing https://github.com/pioarduino/registry/releases/download/0.0.1/scons-4.8.1.zip
INFO Installing https://github.com/pioarduino/registry/releases/download/0.0.1/scons-4.8.1.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: tool-scons@4.40801.0 has been installed!
INFO tool-scons@4.40801.0 has been installed!
Tool Manager: Installing platformio/tool-scons @ ~4.40801.0
INFO Installing platformio/tool-scons @ ~4.40801.0
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: tool-scons@4.40801.0 has been installed!
INFO tool-scons@4.40801.0 has been installed!
INFO Installing tools via idf_tools.py (this may take several minutes)...
Tool Manager: Installing file:///Users/dominik/.platformio/tools/tool-esptoolpy
INFO Installing file:///Users/dominik/.platformio/tools/tool-esptoolpy
Tool Manager: tool-esptoolpy@5.0.2 has been installed!
INFO tool-esptoolpy@5.0.2 has been installed!
INFO Tool tool-esptoolpy successfully installed
INFO Installing tools via idf_tools.py (this may take several minutes)...
Tool Manager: Installing file:///Users/dominik/.platformio/tools/toolchain-xtensa-esp-elf
INFO Installing file:///Users/dominik/.platformio/tools/toolchain-xtensa-esp-elf
Tool Manager: toolchain-xtensa-esp-elf@14.2.0+20241119 has been installed!
INFO toolchain-xtensa-esp-elf@14.2.0+20241119 has been installed!
INFO Tool toolchain-xtensa-esp-elf successfully installed
INFO Installing tools via idf_tools.py (this may take several minutes)...
Tool Manager: Installing file:///Users/dominik/.platformio/tools/tool-mklittlefs
INFO Installing file:///Users/dominik/.platformio/tools/tool-mklittlefs
Tool Manager: tool-mklittlefs@3.2.0 has been installed!
INFO tool-mklittlefs@3.2.0 has been installed!
INFO Tool tool-mklittlefs successfully installed
INFO Installing tools via idf_tools.py (this may take several minutes)...
Tool Manager: Installing file:///Users/dominik/.platformio/tools/tool-cmake
INFO Installing file:///Users/dominik/.platformio/tools/tool-cmake
Tool Manager: tool-cmake@3.30.2 has been installed!
INFO tool-cmake@3.30.2 has been installed!
INFO Tool tool-cmake successfully installed
INFO Installing tools via idf_tools.py (this may take several minutes)...
Tool Manager: Installing file:///Users/dominik/.platformio/tools/tool-ninja
INFO Installing file:///Users/dominik/.platformio/tools/tool-ninja
Tool Manager: tool-ninja@1.13.1 has been installed!
INFO tool-ninja@1.13.1 has been installed!
INFO Tool tool-ninja successfully installed
INFO Installing tools via idf_tools.py (this may take several minutes)...
Tool Manager: Installing file:///Users/dominik/.platformio/tools/tool-esp-rom-elfs
INFO Installing file:///Users/dominik/.platformio/tools/tool-esp-rom-elfs
Tool Manager: tool-esp-rom-elfs@2024.10.11 has been installed!
INFO tool-esp-rom-elfs@2024.10.11 has been installed!
INFO Tool tool-esp-rom-elfs successfully installed
INFO Package configuration completed successfully
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
 - framework-espidf @ 3.50402.0 (5.4.2)
 - tool-cmake @ 3.30.2
 - tool-esp-rom-elfs @ 2024.10.11
 - tool-esptoolpy @ 5.0.2
 - tool-mklittlefs @ 3.2.0
 - tool-ninja @ 1.13.1
 - tool-scons @ 4.40801.0 (4.8.1)
 - toolchain-xtensa-esp-elf @ 14.2.0+20241119
Creating a new virtual environment for IDF Python dependencies
Installing ESP-IDF's Python dependencies
*** Error 1
Reading CMake configuration...
-- git rev-parse returned 'fatal: not a git repository (or any of the parent directories): .git'
-- Building ESP-IDF components for target esp32
-- Configuring incomplete, errors occurred!

fatal: not a git repository (or any of the parent directories): .git
CMake Error at /Users/dominik/.platformio/packages/framework-espidf/tools/cmake/build.cmake:620 (message):
  /Users/dominik/.platformio/penv/.espidf-5.4.2/bin/python: Error while
  finding module specification for 'idf_component_manager.prepare_components'
  (ModuleNotFoundError: No module named 'idf_component_manager')

Call Stack (most recent call first):
  /Users/dominik/.platformio/packages/framework-espidf/tools/cmake/project.cmake:718 (idf_build_process)
  CMakeLists.txt:3 (project)



============================================= [FAILED] Took 127.95 seconds =============================================
```

### Additional information

This also happens with other ESP32 variants, e.g. a C3 SuperMini module using the riscv32-esp toolchain. There an example output would be like this (unfortunately I found no way to deal with the strapping pin warning for the i2 pins on that module):

```
❯ esphome compile /Users/dominik/Documents/_dev/SmartHome/HomeAssistant/_repository/esphome/p7xx-room.yaml
INFO ESPHome 2025.10.2
INFO Reading configuration /Users/dominik/Documents/_dev/SmartHome/HomeAssistant/_repository/esphome/p7xx-room.yaml...
WARNING GPIO8 is a strapping PIN and should only be used for I/O with care.
Attaching external pullup/down resistors to strapping pins can cause unexpected failures.
See https://esphome.io/guides/faq.html#why-am-i-getting-a-warning-about-strapping-pins
WARNING GPIO9 is a strapping PIN and should only be used for I/O with care.
Attaching external pullup/down resistors to strapping pins can cause unexpected failures.
See https://esphome.io/guides/faq.html#why-am-i-getting-a-warning-about-strapping-pins
INFO Generating C++ source...
INFO Integrations changed, cleaning cmake cache...
INFO Deleting /Users/dominik/Documents/_dev/SmartHome/HomeAssistant/_repository/esphome/.esphome/build/p7xx-room/.pioenvs/p72hpkuc-floor/CMakeCache.txt
INFO Compiling app...
Processing p7xx-room (board: esp32-c3-devkitm-1; framework: espidf; platform: https://github.com/pioarduino/platform-espressif32/releases/download/54.03.21-2/platform-espressif32.zip)
------------------------------------------------------------------------------------------------------------------------
INFO Package configuration completed successfully
Tool Manager: Installing https://github.com/pioarduino/registry/releases/download/0.0.1/riscv32-esp-elf-14.2.0_20241119.zip
INFO Installing https://github.com/pioarduino/registry/releases/download/0.0.1/riscv32-esp-elf-14.2.0_20241119.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: toolchain-riscv32-esp@14.2.0+20241119 has been installed!
INFO toolchain-riscv32-esp@14.2.0+20241119 has been installed!
INFO Installing tools via idf_tools.py (this may take several minutes)...
Tool Manager: Installing file:///Users/dominik/.platformio/tools/toolchain-riscv32-esp
INFO Installing file:///Users/dominik/.platformio/tools/toolchain-riscv32-esp
Tool Manager: toolchain-riscv32-esp@14.2.0+20241119 has been installed!
INFO toolchain-riscv32-esp@14.2.0+20241119 has been installed!
INFO Tool toolchain-riscv32-esp successfully installed
INFO Package configuration completed successfully
HARDWARE: ESP32C3 160MHz, 320KB RAM, 4MB Flash
 - framework-espidf @ 3.50402.0 (5.4.2)
 - tool-cmake @ 3.30.2
 - tool-esp-rom-elfs @ 2024.10.11
 - tool-esptoolpy @ 5.0.2
 - tool-mklittlefs @ 3.2.0
 - tool-ninja @ 1.13.1
 - tool-scons @ 4.40801.0 (4.8.1)
 - toolchain-riscv32-esp @ 14.2.0+20241119
Reading CMake configuration...
-- Found Git: /usr/bin/git (found version "2.50.1 (Apple Git-155)")
-- git rev-parse returned 'fatal: not a git repository (or any of the parent directories): .git'
-- The C compiler identification is GNU 14.2.0
-- The CXX compiler identification is GNU 14.2.0
-- The ASM compiler identification is GNU
-- Found assembler: /Users/dominik/.platformio/packages/toolchain-riscv32-esp/bin/riscv32-esp-elf-gcc
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Users/dominik/.platformio/packages/toolchain-riscv32-esp/bin/riscv32-esp-elf-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Users/dominik/.platformio/packages/toolchain-riscv32-esp/bin/riscv32-esp-elf-g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Building ESP-IDF components for target esp32c3
-- Configuring incomplete, errors occurred!

fatal: not a git repository (or any of the parent directories): .git
CMake Error at /Users/dominik/.platformio/packages/framework-espidf/tools/cmake/build.cmake:620 (message):
  /Users/dominik/.platformio/penv/.espidf-5.4.2/bin/python: Error while
  finding module specification for 'idf_component_manager.prepare_components'
  (ModuleNotFoundError: No module named 'idf_component_manager')

Call Stack (most recent call first):
  /Users/dominik/.platformio/packages/framework-espidf/tools/cmake/project.cmake:718 (idf_build_process)
  CMakeLists.txt:3 (project)



============================================= [FAILED] Took 91.62 seconds =============================================
```

The `clean-all` did not change anything, the error message stays the same. When I try to manually set the version to 5.5.1, I still get the same error in the end – but before it also complains about other version mismatches:

```
❯ esphome compile /Users/dominik/Documents/_dev/SmartHome/HomeAssistant/_repository/esphome/p7xx-room.yaml
INFO ESPHome 2025.10.2
INFO Reading configuration /Users/dominik/Documents/_dev/SmartHome/HomeAssistant/_repository/esphome/p7xx-room.yaml...
WARNING The selected framework version is not the recommended one. If there are connectivity or build issues please remove the manual version.
WARNING The selected platform version is not the recommended one. If there are connectivity or build issues please remove the manual version.
WARNING The selected framework version is not the recommended one. If there are connectivity or build issues please remove the manual version.
WARNING The selected platform version is not the recommended one. If there are connectivity or build issues please remove the manual version.
WARNING GPIO8 is a strapping PIN and should only be used for I/O with care.
Attaching external pullup/down resistors to strapping pins can cause unexpected failures.
See https://esphome.io/guides/faq.html#why-am-i-getting-a-warning-about-strapping-pins
WARNING GPIO9 is a strapping PIN and should only be used for I/O with care.
Attaching external pullup/down resistors to strapping pins can cause unexpected failures.
See https://esphome.io/guides/faq.html#why-am-i-getting-a-warning-about-strapping-pins
INFO Generating C++ source...
INFO Compiling app...
Processing p72hpkuc-floor (board: esp32-c3-devkitm-1; framework: espidf; platform: https://github.com/pioarduino/platform-espressif32/releases/download/55.03.31-1/platform-espressif32.zip)
------------------------------------------------------------------------------------------------------------------------
Platform Manager: Installing https://github.com/pioarduino/platform-espressif32/releases/download/55.03.31-1/platform-espressif32.zip
INFO Installing https://github.com/pioarduino/platform-espressif32/releases/download/55.03.31-1/platform-espressif32.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Platform Manager: espressif32@55.3.31 has been installed!
INFO espressif32@55.3.31 has been installed!
INFO Version mismatch for tool-esp_install: installed=5.3.1, required=https://github.com/pioarduino/esp_install/releases/download/v5.3.2/esp_install-v5.3.2.zip, installing correct version
INFO Removing old tool-esp_install installation
INFO Installing tool-esp_install version https://github.com/pioarduino/esp_install/releases/download/v5.3.2/esp_install-v5.3.2.zip
Tool Manager: Installing https://github.com/pioarduino/esp_install/releases/download/v5.3.2/esp_install-v5.3.2.zip
INFO Installing https://github.com/pioarduino/esp_install/releases/download/v5.3.2/esp_install-v5.3.2.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: tool-esp_install@5.3.2 has been installed!
INFO tool-esp_install@5.3.2 has been installed!
INFO tool-esp_install successfully installed and verified
INFO Version mismatch for tool-esptoolpy: 5.0.2 != 5.1.0
INFO Reinstalling tool-esptoolpy due to version mismatch
Created pioarduino Python virtual environment: /Users/dominik/.platformio/penv
INFO Version mismatch for tool-cmake: 3.30.2 != 4.0.3
INFO Reinstalling tool-cmake due to version mismatch
INFO Package configuration completed successfully
Tool Manager: Installing https://github.com/pioarduino/esp-idf/releases/download/v5.5.1/esp-idf-v5.5.1.zip
INFO Installing https://github.com/pioarduino/esp-idf/releases/download/v5.5.1/esp-idf-v5.5.1.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: framework-espidf@3.50501.0 has been installed!
INFO framework-espidf@3.50501.0 has been installed!
Tool Manager: Installing https://github.com/pioarduino/registry/releases/download/0.0.1/esptoolpy-v5.1.0.zip
INFO Installing https://github.com/pioarduino/registry/releases/download/0.0.1/esptoolpy-v5.1.0.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: tool-esptoolpy@5.1.0 has been installed!
INFO tool-esptoolpy@5.1.0 has been installed!
Tool Manager: Installing https://github.com/pioarduino/registry/releases/download/0.0.1/contrib-piohome-3.4.4.tar.gz
INFO Installing https://github.com/pioarduino/registry/releases/download/0.0.1/contrib-piohome-3.4.4.tar.gz
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: contrib-piohome@3.4.4 has been installed!
INFO contrib-piohome@3.4.4 has been installed!
Tool Manager: Installing https://github.com/pioarduino/registry/releases/download/0.0.1/cmake-4.0.3.zip
INFO Installing https://github.com/pioarduino/registry/releases/download/0.0.1/cmake-4.0.3.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Tool Manager: tool-cmake@4.0.3 has been installed!
INFO tool-cmake@4.0.3 has been installed!
INFO Installing tools via idf_tools.py (this may take several minutes)...
Tool Manager: Installing file:///Users/dominik/.platformio/tools/tool-esptoolpy
INFO Installing file:///Users/dominik/.platformio/tools/tool-esptoolpy
Tool Manager: tool-esptoolpy@5.1.0 has been installed!
INFO tool-esptoolpy@5.1.0 has been installed!
INFO Tool tool-esptoolpy successfully installed
INFO Installing tools via idf_tools.py (this may take several minutes)...
Tool Manager: Installing file:///Users/dominik/.platformio/tools/tool-cmake
INFO Installing file:///Users/dominik/.platformio/tools/tool-cmake
Tool Manager: tool-cmake@4.0.3 has been installed!
INFO tool-cmake@4.0.3 has been installed!
INFO Tool tool-cmake successfully installed
INFO Package configuration completed successfully
Installed esptool from tl-install path: /Users/dominik/.platformio/packages/tool-esptoolpy
HARDWARE: ESP32C3 160MHz, 320KB RAM, 4MB Flash
 - contrib-piohome @ 3.4.4
 - framework-espidf @ 3.50501.0 (5.5.1)
 - tool-cmake @ 4.0.3
 - tool-esp-rom-elfs @ 2024.10.11
 - tool-esptoolpy @ 5.1.0
 - tool-mklittlefs @ 3.2.0
 - tool-ninja @ 1.13.1
 - tool-scons @ 4.40801.0 (4.8.1)
 - toolchain-riscv32-esp @ 14.2.0+20241119
Creating a new virtual environment for IDF Python dependencies using uv
Using Python 3.14.0 environment at: /Users/dominik/.platformio/penv/.espidf-5.5.1
Installing ESP-IDF's Python dependencies with uv
Using Python 3.14.0 environment at: /Users/dominik/.platformio/penv/.espidf-5.5.1
Resolved 27 packages in 376ms
Downloading cryptography (6.4MiB)
   Building pydantic-core==2.33.2
 Downloading cryptography
  × Failed to build `pydantic-core==2.33.2`
  ├─▶ The build backend returned an error
  ╰─▶ Call to `maturin.build_wheel` failed (exit status: 1)

      [stdout]
      Running `maturin pep517 build-wheel -i /Users/dominik/.cache/uv/builds-v0/.tmp2pZhUp/bin/python --compatibility
      off`
      cargo 1.90.0 (840b83a10 2025-07-30)
      Rust not found, installing into a temporary directory

      [stderr]
      Python reports SOABI: cpython-314-darwin
      Computed rustc target triple: aarch64-apple-darwin
      Installation directory: /Users/dominik/Library/Caches/puccinialin
      Rustup already downloaded
      Installing rust to /Users/dominik/Library/Caches/puccinialin/rustup
      warn: It looks like you have an existing rustup settings file at:
      warn: /Users/dominik/.rustup/settings.toml
      warn: Rustup will install the default toolchain as specified in the settings file,
      warn: instead of the one inferred from the default host triple.
      info: profile set to 'minimal'
      info: default host triple is aarch64-apple-darwin
      warn: Updating existing toolchain, profile choice will be ignored
      info: syncing channel updates for 'stable-aarch64-apple-darwin'
      info: default toolchain set to 'stable-aarch64-apple-darwin'
      Checking if cargo is installed
      📦 Including license file `LICENSE`
      🍹 Building a mixed python/rust project
      🔗 Found pyo3 bindings
      🐍 Found CPython 3.14 at /Users/dominik/.cache/uv/builds-v0/.tmp2pZhUp/bin/python
      📡 Using build options features, bindings from pyproject.toml
      💻 Using `MACOSX_DEPLOYMENT_TARGET=11.0` for aarch64-apple-darwin by default
         Compiling proc-macro2 v1.0.86
         Compiling unicode-ident v1.0.12
         Compiling target-lexicon v0.13.2
         Compiling python3-dll-a v0.2.13
         Compiling once_cell v1.19.0
         Compiling stable_deref_trait v1.2.0
         Compiling autocfg v1.3.0
         Compiling litemap v0.7.3
         Compiling writeable v0.5.5
         Compiling libc v0.2.155
         Compiling heck v0.5.0
         Compiling icu_locid_transform_data v1.5.0
         Compiling version_check v0.9.5
         Compiling smallvec v1.14.0
         Compiling num-traits v0.2.19
         Compiling icu_properties_data v1.5.0
         Compiling rustversion v1.0.17
         Compiling memoffset v0.9.1
         Compiling static_assertions v1.1.0
         Compiling utf16_iter v1.0.5
         Compiling write16 v1.0.0
         Compiling cfg-if v1.0.0
         Compiling icu_normalizer_data v1.5.0
         Compiling utf8_iter v1.0.4
         Compiling radium v0.7.0
         Compiling memchr v2.7.4
         Compiling quote v1.0.36
         Compiling pyo3-build-config v0.24.1
         Compiling syn v2.0.82
         Compiling lexical-util v1.0.6
         Compiling ahash v0.8.11
         Compiling serde v1.0.218
         Compiling tap v1.0.1
         Compiling wyz v0.5.1
         Compiling num-integer v0.1.46
         Compiling getrandom v0.2.15
         Compiling aho-corasick v1.1.3
         Compiling num-bigint v0.4.6
         Compiling zerocopy v0.7.34
         Compiling lexical-parse-integer v1.0.5
         Compiling funty v2.0.0
         Compiling regex-syntax v0.8.5
         Compiling percent-encoding v2.3.1
         Compiling indoc v2.0.5
         Compiling hashbrown v0.14.5
         Compiling unindent v0.2.3
         Compiling equivalent v1.0.1
         Compiling serde_json v1.0.140
         Compiling bitvec v1.0.1
         Compiling pyo3-ffi v0.24.1
         Compiling pyo3-macros-backend v0.24.1
         Compiling pyo3 v0.24.1
         Compiling jiter v0.9.0
      error: failed to run custom build command for `pyo3-ffi v0.24.1`

      Caused by:
        process didn't exit successfully:
      `/Users/dominik/.cache/uv/sdists-v9/pypi/pydantic-core/2.33.2/6Jc22htzrjzwICc-p5aFK/src/target/release/build/pyo3-ffi-3a915bcb0b4273f6/build-script-build`
      (exit status: 1)
        --- stdout
        cargo:rustc-check-cfg=cfg(Py_LIMITED_API)
        cargo:rustc-check-cfg=cfg(Py_GIL_DISABLED)
        cargo:rustc-check-cfg=cfg(PyPy)
        cargo:rustc-check-cfg=cfg(GraalPy)
        cargo:rustc-check-cfg=cfg(py_sys_config, values("Py_DEBUG", "Py_REF_DEBUG", "Py_TRACE_REFS", "COUNT_ALLOCS"))
        cargo:rustc-check-cfg=cfg(pyo3_disable_reference_pool)
        cargo:rustc-check-cfg=cfg(pyo3_leak_on_drop_without_reference_pool)
        cargo:rustc-check-cfg=cfg(Py_3_7)
        cargo:rustc-check-cfg=cfg(Py_3_8)
        cargo:rustc-check-cfg=cfg(Py_3_9)
        cargo:rustc-check-cfg=cfg(Py_3_10)
        cargo:rustc-check-cfg=cfg(Py_3_11)
        cargo:rustc-check-cfg=cfg(Py_3_12)
        cargo:rustc-check-cfg=cfg(Py_3_13)
        cargo:rustc-check-cfg=cfg(Py_3_14)
        cargo:rerun-if-env-changed=PYO3_CROSS
        cargo:rerun-if-env-changed=PYO3_CROSS_LIB_DIR
        cargo:rerun-if-env-changed=PYO3_CROSS_PYTHON_VERSION
        cargo:rerun-if-env-changed=PYO3_CROSS_PYTHON_IMPLEMENTATION
        cargo:rerun-if-env-changed=PYO3_PRINT_CONFIG
        cargo:rerun-if-env-changed=PYO3_USE_ABI3_FORWARD_COMPATIBILITY

        --- stderr
        error: the configured Python interpreter version (3.14) is newer than PyO3's maximum supported version (3.13)
        = help: please check if an updated version of PyO3 is available. Current version: 0.24.1
        = help: set PYO3_USE_ABI3_FORWARD_COMPATIBILITY=1 to suppress this check and build anyway using the stable ABI
      warning: build failed, waiting for other jobs to finish...
      💥 maturin failed
        Caused by: Failed to build a native library through cargo
        Caused by: Cargo build finished with "exit status: 101": `env -u CARGO MACOSX_DEPLOYMENT_TARGET="11.0"
      PYO3_BUILD_EXTENSION_MODULE="1" PYO3_ENVIRONMENT_SIGNATURE="cpython-3.14-64bit"
      PYO3_PYTHON="/Users/dominik/.cache/uv/builds-v0/.tmp2pZhUp/bin/python"
      PYTHON_SYS_EXECUTABLE="/Users/dominik/.cache/uv/builds-v0/.tmp2pZhUp/bin/python" "cargo" "rustc"
      "--features" "pyo3/extension-module" "--message-format" "json-render-diagnostics" "--manifest-path"
      "/Users/dominik/.cache/uv/sdists-v9/pypi/pydantic-core/2.33.2/6Jc22htzrjzwICc-p5aFK/src/Cargo.toml"
      "--release" "--lib" "--crate-type" "cdylib" "--" "-C" "link-arg=-undefined" "-C" "link-arg=dynamic_lookup" "-C"
      "link-args=-Wl,-install_name,@rpath/pydantic_core._pydantic_core.cpython-314-darwin.so"`
      Error: command ['maturin', 'pep517', 'build-wheel', '-i',
      '/Users/dominik/.cache/uv/builds-v0/.tmp2pZhUp/bin/python', '--compatibility', 'off'] returned non-zero exit
      status 1

      hint: This usually indicates a problem with the package or the build environment.
  help: `pydantic-core` (v2.33.2) was included because `pydantic` (v2.11.10) depends on `pydantic-core`
*** Error 1
Reading CMake configuration...
-- Found Git: /usr/bin/git (found version "2.50.1 (Apple Git-155)")
-- git rev-parse returned 'fatal: not a git repository (or any of the parent directories): .git'
-- Minimal build - OFF
-- The C compiler identification is GNU 14.2.0
-- The CXX compiler identification is GNU 14.2.0
-- The ASM compiler identification is GNU
-- Found assembler: /Users/dominik/.platformio/packages/toolchain-riscv32-esp/bin/riscv32-esp-elf-gcc
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Users/dominik/.platformio/packages/toolchain-riscv32-esp/bin/riscv32-esp-elf-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Users/dominik/.platformio/packages/toolchain-riscv32-esp/bin/riscv32-esp-elf-g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Building ESP-IDF components for target esp32c3
-- Configuring incomplete, errors occurred!

fatal: not a git repository (or any of the parent directories): .git
CMake Error at /Users/dominik/.platformio/packages/framework-espidf/tools/cmake/build.cmake:629 (message):
  /Users/dominik/.platformio/penv/.espidf-5.5.1/bin/python: Error while
  finding module specification for 'idf_component_manager.prepare_components'
  (ModuleNotFoundError: No module named 'idf_component_manager')

Call Stack (most recent call first):
  /Users/dominik/.platformio/packages/framework-espidf/tools/cmake/project.cmake:740 (idf_build_process)
  CMakeLists.txt:3 (project)



============================================= [FAILED] Took 99.34 seconds =============================================
```

It seems to take the python version 3.14 from Homebrew, which breaks some of the dependencies.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -19,6 +19,8 @@ classifiers = [
     "Programming Language :: Python :: 3",
     "Topic :: Home Automation",
 ]
+
+# Python 3.14 is currently not supported by IDF <= 5.5.1, see https://github.com/esphome/esphome/issues/11502
 requires-python = ">=3.11.0,<3.14"
 
 dynamic = ["dependencies", "optional-dependencies", "version"]

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -19,7 +19,7 @@ classifiers = [
     "Programming Language :: Python :: 3",
     "Topic :: Home Automation",
 ]
-requires-python = ">=3.11.0"
+requires-python = ">=3.11.0,<3.14"
 
 dynamic = ["dependencies", "optional-dependencies", "version"]
 

